### PR TITLE
Ormichae/apprentice aggregation

### PR DESCRIPTION
**Pull request by orenmichaely**
Updated the apprentice mode aggregation in the external parser.

**Old Apprentice Mode reward signal:**  

  If personalizer matches baseline: 
     Baseline reward is passed (if the baseline system receives a negative reward, the system receives a negative reward even though it successfully matched baseline) 
  
  If personalizer does not match baseline 
      Default reward 

**New reward signal** 

  If personalizer matches baseline 
      Reward of 1 
  
  If personalizer does not match baseline 
      reward of 0

### CODE REVIEW

File: external_parser/event_processors/joined_event.h [hunk 1]
```diff
@@ -77,9 +77,12 @@ struct cb_joined_event : public typed_joined_event {
     if (!interaction_data.actions.empty() &&
         interaction_data.actions[0] == CB_BASELINE_ACTION)
     {
-      // TODO: default apprenticeReward should come from config
-      // setting to default reward matches current behavior for now
-      reward = original_reward;
+      // The goal of Apprentice mode is to successfully imitate the baseline system.
+      // Reward of 1 when we are match baseline and 0 otherwise
+      reward = 1.0;
```
Comment (ataymano) [inline external_parser/event_processors/joined_event.h:82]: can we define constants for these 0/1 values?

File: external_parser/event_processors/joined_event.h [hunk 2]
```diff
@@ -43,6 +43,8 @@ inline void calculate_multislot_interaction_metrics(dsjson_metrics* metrics,
   }
 
 struct typed_joined_event {
+  const float apprentice_matching_reward = 1.0f;
```
Comment (ataymano) [inline external_parser/event_processors/joined_event.h:46]: can be static

Comment (orenmichaely) [reply external_parser/event_processors/joined_event.h]: agreed:)

**Review by ataymano — APPROVED** [review]

File: external_parser/unit_tests/test_vw_external_parser.cc [hunk 3]
```diff
@@ -1,6 +1,7 @@
 #include "ccb_label.h"
 #include "test_common.h"
 
+#include <boost/test/tools/old/interface.hpp>
```
Comment (olgavrou) [inline external_parser/unit_tests/test_vw_external_parser.cc:4]: nit: do we need this include?

**Review by olgavrou — APPROVED** [review]

**Pull request merged**